src/Symfony/Component/AssetMapper/ImportMap/ImportMapAuditor.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Component\AssetMapper\Exception\RuntimeException;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImportMapAuditor
+{
+    private const AUDIT_URL = 'https://registry.npmjs.org/-/npm/v1/security/audits';
+
+    private readonly HttpClientInterface $httpClient;
+
+    public function __construct(
+        private readonly string $importMapConfigPath,
+        HttpClientInterface $httpClient = null,
+    ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
+    }
+
+    public function audit(): array
+    {
+        $packages = $this->getPackagesVersion();
+
+        if ([] === $packages) {
+            return [];
+        }
+
+        $response = $this->httpClient->request('POST', self::AUDIT_URL, [
+            'json' => [
+                'requires' => array_map(fn(): string => '*', $packages),
+                'dependencies' => array_map(fn(string $version): array => ['version' => $version], $packages),
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new RuntimeException(sprintf('Error %d auditing packages. Response: %s', $response->getStatusCode(), $response->getContent(false)));
+        }
+
+
+        $data = $response->toArray();
+
+        return [
+            'packages' => $packages,
+            'advisories' => $data['advisories'],
+            'summary' => $data['metadata']['vulnerabilities'],
+        ];
+    }
+
+    private function getPackagesVersion(): array
+    {
+        $path = $this->importMapConfigPath;
+        $importMapConfig = is_file($path) ? (static fn () => include $path)() : [];
+
+        $packages = [];
+        foreach ($importMapConfig as $data) {
+            if (null === $url = $data['url'] ?? null) {
+                continue;
+            }
+
+            // try to grab the version from the URL
+            $registryRegex = '(?<registry>\w+)';
+            $packageRegex = '(?<package>(?:@[a-z0-9-~][a-z0-9-._~]*/)?[a-z0-9-~][a-z0-9-._~]*)';
+            $versionRegex = '(?<version>[\w\._-]+)';
+            $subPathRegex = '(?<subpath>/.*)';
+
+            if (
+                1 === preg_match("#^https://ga.jspm.io/{$registryRegex}:{$packageRegex}@{$versionRegex}{$subPathRegex}?$#", $url, $matches)
```

Comment:
why using a registry regex when the only valid value for something supported by the npmjs endpoint would be `npm` ?

For instance, if you use the `gh` ecosystem for cdn.jsdelivr.net URLs, you won't get an npm package name after it, so this won't be valid.

Reply:
This logic has been moved to resolvers. `JsDelivrEsmResolver` already get the version when requiring a package. `JspmResolver` still use a regex (which might be improved I think).